src/prommis/uky/uky_flowsheet.py

```diff
@@ -861,6 +871,733 @@ def solve(m):
 def display_results(m):
     m.fs.roaster.display()
 
+    # precipitate = value(units.convert(m.fs.leach_filter_cake.flow_mass[0], to_units=units.kg / units.hr))
+    # print(f"Precipitate mass flow is {precipitate} kg/hr")
+    # product = value(units.convert(m.fs.roaster.flow_mas_product[0], to_units=units.kg / units.hr))
+    # print(f"REE product mass flow is {product} kg/hr")
+    # dust = value(units.convert(m.fs.roaster.flow_mas_dust[0], to_units=units.kg / units.hr))
+    # print(f"Dust mass flow is {dust} kg/hr")
+    #
+    # precipitate_percent = 100 * precipitate / (precipitate + product + dust)
+    # product_percent = 100 * product / (precipitate + product + dust)
+    # dust_percent = 100 * dust / (precipitate + product + dust)
+    # print(f"Precipitate is {precipitate_percent}% of total outlet mass flow")
+    # print(f"Product is {product_percent}% of total outlet mass flow")
+    # print(f"Dust is {dust_percent}% of total outlet mass flow")
+
+
+def add_costing(flowsheet):
+    m = ConcreteModel()
+
+    # Add a flowsheet object to the model
+    m.fs = FlowsheetBlock(dynamic=True, time_units=units.s)
+    m.fs.costing = QGESSCosting()
+    CE_index_year = "UKy_2019"
+
+    # Leaching costs
+    # 4.2 is UKy Leaching - Polyethylene Tanks
+    L_pe_tanks_accounts = ["4.2"]
+    m.fs.L_pe_tanks = UnitModelBlock()
+    m.fs.L_pe_tanks.capacity = Var(
+        initialize=value(flowsheet.fs.leach.volume[0, 2]), units=units.gal
+    )
+    m.fs.L_pe_tanks.capacity.fix()
+    m.fs.L_pe_tanks.costing = UnitModelCostingBlock(
+        flowsheet_costing_block=m.fs.costing,
+        costing_method=QGESSCostingData.get_REE_costing,
+        costing_method_arguments={
+            "cost_accounts": L_pe_tanks_accounts,
+            "scaled_param": m.fs.L_pe_tanks.capacity,
+            "source": 1,
+            "n_equip": flowsheet.fs.leach.config.number_of_finite_elements,
+            "scale_down_parallel_equip": False,
+            "CE_index_year": CE_index_year,
+        },
+    )
+
+    # 4.3 is UKy Leaching - Tank Mixer
+    L_tank_mixer_accounts = ["4.3"]
+    m.fs.L_tank_mixers = UnitModelBlock()
+    m.fs.L_tank_mixers.power = Var(initialize=474, units=units.hp)
```

Comment:
I've updated hard-coded values where possible, but there are still a lot of values that we don't (to my knowledge) have values for in the flowsheet. For instance, what should this horsepower be? The hard-coded numbers seem to represent a system that is orders of magnitude larger than this system, so most of these values will likely need to be changed.